src/python/pants/backend/jvm/tasks/junit_run.py

```diff
@@ -644,23 +661,46 @@ def _isolation(self, all_targets):
     try:
       yield output_dir, reports, coverage
     finally:
-      # NB: Deposit of the "current" test output in the root workdir (.pants.d/test/junit) is a
-      # defacto public API and so we implement that behavior here to maintain backwards
-      # compatibility for non-pants report file consumers.
-      # TODO(John Sirois): Deprecate this ~API and provide a stable directory solution for test
-      # output: https://github.com/pantsbuild/pants/issues/3879
       lock_file = '.file_lock'
       with OwnerPrintingInterProcessFileLock(os.path.join(self.workdir, lock_file)):
-        # Kill everything except the isolated `_runs/` dir.
-        for name in os.listdir(self.workdir):
-          path = os.path.join(self.workdir, name)
-          if name not in (run_dir, lock_file):
-            if os.path.isdir(path):
-              safe_rmtree(path)
-            else:
-              os.unlink(path)
+        preserve = (run_dir, lock_file)
+        self._link_current_reports(output_dir, preserve)
+
+  def _link_current_reports(self, output_dir, preserve):
+    # NB: Deposit of the "current" test output in the root workdir (.pants.d/test/junit) is a
+    # defacto public API and so we implement that behavior here to maintain backwards
+    # compatibility for non-pants report file consumers.
+    # TODO(John Sirois): Deprecate this ~API and provide a stable directory solution for test
+    # output: https://github.com/pantsbuild/pants/issues/3879
+
+    # Kill everything except the isolated `_runs/` dir.
+    for name in os.listdir(self.workdir):
+      path = os.path.join(self.workdir, name)
+      if name not in preserve:
+        if os.path.isdir(path):
+          safe_rmtree(path)
+        else:
+          os.unlink(path)
+
+    # Link ~all the isolated run/ dir contents back up to the stable workdir
+    # NB: In the case of like-named files, which can be emitted under different subdirs when
+    # batching is enabled, the last file found will win. We accept this truncation of the full
+    # data since this feature is on the road to deprecation, and since prior to batch fixes to
+    # preserve batch report information using separate batch output subdirs, a last-report-wins
+    # semantic was in-place on the generation side of things.
+    for root, dirs, files in safe_walk(output_dir, topdown=True):
```

Comment:
I found this comment a little tricky to follow. I tried rewording it below. Does this mean the same thing?

When batching is enabled, files can be emitted under different subdirs. If those files have like-names, the last file with a like-name will be the one that is used. This may result in a loss of information from the ignored files.
We're OK with this because a) we're planning on deprecating this loss of information and b) it is the same behavior as existed before batching was added.

Reply:
It does mean the same thing, took your wording.